emcc.py

```diff
@@ -1741,6 +1741,9 @@ def phase_linker_setup(options, state, newargs, user_settings):
   # Note the exports the user requested
   building.user_requested_exports.update(settings.EXPORTED_FUNCTIONS)
 
+  if '_main' in settings.EXPORTED_FUNCTIONS:
+    settings.EXPORT_IF_DEFINED.append('__main_argc_argv')
```

Comment:
A comment here would be good. This stuff is adding a bunch of complexity and comments can at least help the reader with that. The other locations at least use the `MANGLED_MAIN` setting that the reader can then look up, but this location doesn't even have that.